front/src/modules/timesStops/helpers/scheduleData.ts

```diff
@@ -7,31 +8,31 @@ import type { ComputedScheduleEntry, ScheduleEntry } from '../types';
  * @param schedule for a given operational point
  * @param startTime time of departure from the very beginning of the route
  */
-export function computeScheduleData(schedule: ScheduleEntry, startTime: string) {
+export function computeScheduleData(schedule: ScheduleEntry) {
   if (!schedule) {
     return { arrival: null, departure: null, stopFor: null };
   }
-  const startTimeSeconds = datetime2sec(new Date(startTime));
-  // relative value, number of seconds since startTime
-  const arrivalSeconds = schedule.arrival
-    ? startTimeSeconds + ISO8601Duration2sec(schedule.arrival)
-    : null;
-
+  const arrivalSeconds = schedule.arrival ? ISO8601Duration2sec(schedule.arrival) : null;
   const stopForSeconds = schedule.stop_for ? ISO8601Duration2sec(schedule.stop_for) : null;
+  const departureSeconds =
+    arrivalSeconds && stopForSeconds ? arrivalSeconds + stopForSeconds : null;
 
-  const departure =
-    arrivalSeconds && stopForSeconds ? startTimeSeconds + arrivalSeconds + stopForSeconds : null;
   return {
     arrival: arrivalSeconds,
-    departure,
+    departure: departureSeconds,
     stopFor: stopForSeconds,
   };
 }
 
-export function formatScheduleData(scheduleData: ComputedScheduleEntry) {
+export function formatScheduleData(
+  scheduleData: ComputedScheduleEntry
+): Pick<SuggestedOP, 'arrival' | 'departure' | 'stopFor'> {
+  const arrival = scheduleData.arrival ? formatDurationAsISO8601(scheduleData.arrival) : null;
+  const departure = scheduleData.departure ? formatDurationAsISO8601(scheduleData.departure) : null;
```

Comment:
To keep track of the kind of time we use

```suggestion
  const arrival: IsoDurationString | null = scheduleData.arrival
    ? formatDurationAsISO8601(scheduleData.arrival)
    : null;
  const departure: IsoDurationString | null = scheduleData.departure
    ? formatDurationAsISO8601(scheduleData.departure)
    : null;
```

Reply:
done